src/vs/platform/extensions/node/extensionHostStarter.ts

```diff
@@ -115,17 +122,32 @@ class ExtensionHostProcess extends Disposable {
 		this._logService.info(`Killing extension host with pid ${this._process.pid}.`);
 		this._process.kill();
 	}
+
+	async waitForExit(maxWaitTimeMs: number): Promise<void> {
+		if (!this._process) {
+			return;
+		}
+		const pid = this._process.pid;
+		this._logService.info(`Waiting for extension host with pid ${pid} to exit.`);
+		await Promise.race([Event.toPromise(this.onExit), timeout(maxWaitTimeMs)]);
```

Comment:
To clarify: we are not explicitly terminating the extension host here because we assume that the window is closing which will close the socket and then should trigger the self-termination from within the extension host?

I wonder if maybe a more explicit flow should be implemented where you terminate the extension host from here, because once we move to a different IPC mechanism in the future, maybe we cannot rely on the socket-close event anymore.

Reply:
Yes, we need to revisit the shutdown of the extension host. Today, the extension host shuts down when the renderer sends it a shutdown message via the socket or when the socket closes. If the new IPC mechanism does not support closing we need to find an alternative way to shutdown the extension host (maybe by posting a message using nodejs to it).